samples/hello-cube.html

```diff
@@ -0,0 +1,313 @@
+<!DOCTYPE html>
+<html>
+<head>
+<meta name="viewport" content="width=600">
+<title>Web GPU Cube demo</title>
+<script src="gl-matrix-min.js"></script>
+</head>
+<body>
+<canvas height=1000 width=1000></canvas>
+<script>
+/*******
+* This preview of Web GPU uses WHLSL as its shading language. 
+* The choice of shading language(s) that will be ingested by the API is still under deliberation. 
+*******/
+const shader = `
+struct FragmentData {
+    float4 position : SV_Position;
+    float4 color : attribute(0);
+};
+
+vertex FragmentData vertex_main(float4 position : attribute(0),
+                                float4 color : attribute(1),
+                                constant float4x4* modelViewProjectionMatrix : register(b0)) {
+    FragmentData output;
+
+    output.position = *modelViewProjectionMatrix * position;
+    output.color = color;
+
+    return output;
+}
+
+fragment float4 fragment_main(float4 color : attribute(0)) : SV_Target {
+    return color;
+}
+`
+
+const vertexSize = 4 * 8;
+const colorOffset = 4 * 4;
+const vertexArray = new Float32Array([
```

Comment:
I have no idea whether this giant array is correct. Can't you compute it at runtime?

Reply:
This is fairly standard in all spinning cube demos. I don't think it is a big deal. Computing it at runtime would probably be an equally confusing set of for loops.

Maybe the best thing to do would be to name the variables cubeVertices, and provide comments for the 4 * 8 and 4 * 4 values.